### DISCUSSION TITLE
Post Processor for a specific robot/CNC controller

### DISCUSSION THREAD

**Discussion by VitaliiSadovshikov** [fellesverkstedet/Bark-beetle-parametric-toolpaths#84] [General]
Hi there, at first glance, I have noticed that your super projectile spit out G-code and SBP-code. But let's ask you a naive question is it possible here to apply trademark postprocessor? I worry that the conventional g-code won't work for some CNC machines and especially with a robot arm.

**Comment by JensDyvik** [comment]
That is absolutely possible. You can modify the post processor component to format whatever language you need. It is actually programmed with only regular grasshopper components.

If you double click one of the post processor clusters you can modify the logic inside:
![image](https://user-images.githubusercontent.com/6598148/154082739-e6fa5358-546c-4a8b-a744-9af8c7fe9a0d.png)

Once inside you can see the crude but effective way of converting polyline coordinates into g-code:
![image](https://user-images.githubusercontent.com/6598148/154083337-bdd0c84f-e608-4263-9521-95fdb956773d.png)

If you change any of the post processor clusters and save, all of them will update.

About robot arms, a few years ago made a small experiment making six axis milling toolpaths and formating them to output "Inform" language code for  a 1997 Motoman industrial robot. It might be a good reference for you if you want to make custom CAM for robot arms. The grasshopper files have ended up here (I need to do some house keeping): https://github.com/fellesverkstedet/Bark-beetle-parametric-toolpaths/blob/master/Archive/Bark%20beetle%20components%20-%20Inkscape/Examples%20-%20Latest%20development/Bark%20beetle%20-%20Six%20axis%20robot%20milling.gh

https://github.com/fellesverkstedet/Bark-beetle-parametric-toolpaths/blob/master/Archive/Bark%20beetle%20components%20-%20Inkscape/Examples%20-%20Latest%20development/Old%20experiments/Bark%20beetle%20-%20Robofab.gh

Here is a video of the last gh file in action: https://youtu.be/GnwZuvjI4-o

**Reply by VitaliiSadovshikov** [reply]
Much obliged! Should dig in. Cool work by the way

**Comment by JensDyvik** [comment]
Thank you! Please post some pics and files here if when you get some results :)
